src/components/utils/src/push_log.cc

```diff
@@ -41,7 +41,24 @@ struct __attribute__((visibility("default"))) LogsEnabled {
 };
 
 std::atomic_bool LogsEnabled::logs_enabled_(false);
-static std::unique_ptr<LogMessageLoopThread> log_message_loop_thread;
+
+template <typename T>
+using logger_ptr = std::unique_ptr<T, std::function<void(T*)> >;
+
+static logger_ptr<LogMessageLoopThread> log_message_loop_thread;
+
+auto deinit_logger = [](LogMessageLoopThread* logMsgThread) {
+  delete logMsgThread;
+  logger::logger_status = logger::LoggerThreadNotCreated;
+
+  log4cxx::LoggerPtr logger(log4cxx::Logger::getLogger("Utils"));
```

Comment:
@SNiukalov why utils? Maybe it's possible to get a logger ptr from input parameter `logMsgThread`? It might be any logger

Reply:
@AKalinich-Luxoft 
Hi, thanks to question I will try explain.

Each plugin has its own logger object, which is created when the plugin is loaded, this logger object must be deleted when the plugin is closed.
Removing logger  occurs in the address space of the plugin
In this case, the plugin only has access to its own logger.

Utils, because, this component is statically linked to the plug-in and is responsible for the log